src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb

```diff
@@ -96,10 +96,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End If
 
             If node.RelaxationLambdaOpt IsNot Nothing Then
-                returnValue = node.Update(VisitExpressionNode(node.RelaxationLambdaOpt),
+
+                If _inExpressionLambda AndAlso
+                    NoParameterRelaxation(DirectCast(node.Operand, BoundLambda).LambdaSymbol, node.RelaxationLambdaOpt.LambdaSymbol) Then
```

Comment:
Something like this:
```
    Sub Test()
        Dim anonymousDelegate = Function(x As Integer) x
        Dim lambda As Expressions.Expression(Of Action) = Sub() M1(anonymousDelegate)
    End Sub

    Sub M1(y As Action(Of Integer))
    End Sub
```

---
In reply to: [64994524](https://github.com/dotnet/roslyn/pull/11627#discussion_r64994524) [](ancestors = 64994524,64994396)